server/src/main/java/app/config/JwtService.java

```diff
@@ -79,4 +87,15 @@ private Key getSignInKey() {
         byte[] keyBytes = Decoders.BASE64.decode(SECRET_KEY);
         return Keys.hmacShaKeyFor(keyBytes);
     }
+
+    public boolean hasAccess(Authentication authentication, Integer userId) {
+        String email = authentication.getName();
+        User currentUser = userRepository.findByEmail(email).orElseThrow(() -> new RuntimeException("User not found"));
+
+        if (currentUser.getRole() == Role.ADMIN) {
+            return true;
+        }
+
+        return currentUser.getId().equals(userId) || friendshipService.isFriend(currentUser.getId(), userId);

```

Comment:
imo to są różne acessy 

rozumiem że zrobiłaś segregacje za pomocą isSame na frontendzie, ale tu wychodzi że moji znajomi za pomocą api mogą modyfikować moje activity co nie jest prawdą, imo tu powinno być dwie funckji z różnymi dostępami